### PR TITLE
slow reducer

### PR DESCRIPTION
**Pull request by jyotima1**

### CODE REVIEW

**Comment by confusionblinds** [comment]
could you stick the following link next to "Open beeline"
https://docs.microsoft.com/en-us/azure/hdinsight/hadoop/apache-hadoop-use-hive-beeline
If possible past a screen of where to find "the value of set hive.optimize.sort.dynamic.partition"."

File: hive/hive-slow-reducer.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+---
+
+### Why is my reducer slow?
+
+#### Issue:
+I am running a select and insert on a large dataset. The data in the datset is skewed and most of the data resides in one partition.
+
+When I run a simple query which looks like this:
+```insert into table1 select a,b,c from table2;```
```
Comment (ashitgosalia) [inline hive/hive-slow-reducer.md]: Is this the correct query ?

File: hive/hive-slow-reducer.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+the query plan starts a bunch of reducers but the data from the heaviest partition goes to a single reducer, causing it to run for a long time.
```
Comment (ashitgosalia) [inline hive/hive-slow-reducer.md]: How can a user identify this ? Is it only a single partitions or can it be a few ?

File: hive/hive-slow-reducer.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+#### Troubleshoot: 
+Open beeline and verify the value of 
+```set hive.optimize.sort.dynamic.partition```
+
+If the value of the parameter is set to true, it means that dynamic partitioning column will be globally sorted. This way we can keep only one record writer open for each partition value in the reducer thereby reducing the memory pressure on reducers.
+
+#### Resolution: 
+1. Try to repartition the data to normalize into multiple partitions.
```
Comment (ashitgosalia) [inline hive/hive-slow-reducer.md]: What does this mean ?

File: hive/hive-slow-reducer.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+2. If #1 is not possible, set the value of the config to false in beeline session and try the query again.
+```set hive.optimize.sort.dynamic.partition=false```.
+Setting the value to false might cause Out of Memory errors. 
+The effect of setting the value to false at a cluster level in a HDP cluster is still being investigated. In the meantime please use the setting only for affected slow queries.
```
Comment (ashitgosalia) [inline hive/hive-slow-reducer.md]: Why refer to HDP here ?

Comment (jyotima1) [reply hive/hive-slow-reducer.md]: corrected this.

File: hive/hive-slow-reducer.md (hunk 3 above)
Comment (jyotima1) [reply hive/hive-slow-reducer.md]: changed the explanation to make more sense now.

**Comment by confusionblinds** [comment]
Adding hive-Slow-Reducer

**Pull request merged**